### PR TITLE
Migrate `run-make/rustdoc-map-file` to rmake

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Fixed formatting.

File: tests/run-make/rustdoc-map-file/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    rustdoc()
+        .input("foo.rs")
+        .arg("-Zunstable-options")
+        .arg("--generate-redirect-map")
+        .output(&out_dir)
+        .run();
+    let python = std::env::var("PYTHON").unwrap_or("python".into());
+    assert!(
+        Command::new(python).arg("validate_json.py").arg(&out_dir).status().unwrap().success()
+    );
```
Comment (jieyouxu) [inline tests/run-make/rustdoc-map-file/rmake.rs:15]: This is very amusing to me. Let me go check what `validate_json.py` does.

Comment (jieyouxu) [reply tests/run-make/rustdoc-map-file/rmake.rs:15]: https://github.com/rust-lang/rust/blob/e3029d220f6843f9b83d35927ff20a86137d7ed0/tests/run-make/rustdoc-map-file/validate_json.py

**Review by jieyouxu — APPROVED** [review]
The ported test looks reasonable to me. The Python script feels a bit wonky as an external dependency but the ported test merely preserves the behavior of the existing test.

I'm okay with not porting the Python script over for now, but imo we should leave a FIXME in that case.

Feel free to r=me after either:

- porting the `validate_json.py` script over, or
- adding a FIXME to remind porting the Python script over as well.

**Comment by GuillaumeGomez** [comment]
I added the FIXME for the time being. When I'm back from the rustNL conference, I'll port the python script as well.

**Comment by jieyouxu** [comment]
Thanks, feel free to r=me after CI is green 💚.

**Pull request merged**